### ISSUE TITLE
serious impute bug introduced in CRAN version 2.9.2

### ISSUE THREAD

**Issue by sheffe** [kogalur/randomForestSRC#59]
Just starting to use this package, but already a huge fan. Thanks for putting it out there!

I have a reproducible example of what appears to be a serious bug in mForest imputation that was introduced in the version change from 2.9.1 to 2.9.2. I've included test code below and a few screenshots of the results.

Unfortunately I don't have any proactive suggestions for what could be causing this, but I'll try to dig in later this week or over the weekend.

```r
library(tidyverse)
library(randomForestSRC)

test_rows <- 10000

set.seed(1234)
fake_data <- dplyr::tibble(
  a = rpois(test_rows, 500),
  b = a + rnorm(test_rows, 50, 50),
  c = b + rnorm(test_rows, 50, 50),
  d = c + rnorm(test_rows, 50, 50),
  e = d + rnorm(test_rows, 50, 50),
  f = e + rnorm(test_rows, 50, 50),
  g = f + rnorm(test_rows, 50, 50),
  h = g + rnorm(test_rows, 50, 50),
  i = h + rnorm(test_rows, 50, 50)
)

fake_data_missing <- fake_data %>%
  dplyr::mutate_if(is.numeric,
                   list(~ifelse(runif(test_rows) <= 0.1, NA, .)))

cor(fake_data)

imputed_data <- randomForestSRC::impute(
  formula = a ~ b + c + d + e + f + g + h + i,
  data = as.data.frame(fake_data_missing),
  mf.q = 0.2,
  ntree = 100,
  verbose = TRUE
)

fake_long <- fake_data %>%
  dplyr::mutate(row_n = row_number()) %>%
  tidyr::gather(key = feature, value = rawval, -row_n)

imputed_long <- imputed_data %>%
  dplyr::mutate(row_n = row_number()) %>%
  tidyr::gather(key = feature, value = imputedval, -row_n)

dplyr::full_join(fake_long, imputed_long, by = c("row_n", "feature")) %>%
  # drop out cases of exact equality
  dplyr::filter(rawval != imputedval) %>%
  ggplot(data = ., aes(x = rawval, y = imputedval)) +
  facet_wrap(~feature, scales = "free") +
  geom_point(size = 0.5) +
  geom_smooth()
```

Using version 2.9.1 this has excellent performance, as we'd expect:
<img width="1524" alt="Screenshot 2020-01-14 16 24 46" src="https://user-images.githubusercontent.com/3778733/72385032-a2384580-36ec-11ea-9c0a-0c60a7cee116.png">

Using version 2.9.2 on the identical data, it breaks down entirely:
<img width="1412" alt="Screenshot 2020-01-14 16 25 58" src="https://user-images.githubusercontent.com/3778733/72385091-be3be700-36ec-11ea-9ce9-02ef86e0a2d0.png">

**Comment by kogalur** [comment]
Thank you!  We have confirmed the issue and will update you soon.

**Comment by kogalur** [comment]
This has been fixed in 2.9.3.  Thank you.
